libs/tree-sitter-wing/grammar.js

```diff
@@ -511,14 +511,34 @@ module.exports = grammar({
 
     access_modifier: ($) => choice("public", "private", "protected"),
 
+    variadic: ($) => "...",
+
+    variadic_container_type: ($) => seq(field("collection_type", choice("Array")), $._container_value_type),
+
+    variadic_definition: ($) => seq(
+      field("variadic", $.variadic),
+      field("name", $.identifier),
+      seq(":", field("type", $.variadic_container_type)),
+    ),
+
     parameter_definition: ($) =>
       seq(
         optional(field("reassignable", $.reassignable)),
         field("name", $.identifier),
         $._type_annotation
       ),
 
-    parameter_list: ($) => seq("(", commaSep($.parameter_definition), ")"),
+    parameter_list: ($) => seq(
+      "(",
+      choice(
+        seq(
+          commaSep($.parameter_definition),
+          optional(seq(",", $.variadic_definition)),
+        ),
+        $.variadic_definition,
+      ),
+      ")"
+    ),
```

Comment:
Even though it makes the grammar a bit less precise, I think we can improve the error diagnostics if we specify in the grammar any parameter can be variadic, and then handle the validation that only the last parameter can be variadic inside `parser.rs`. This way we can give a more friendly error than "unexpected variadic definition".

Something like this maybe?

 ```
    parameter_definition: ($) =>
      seq(
        optional(field("reassignable", $.reassignable)),
        optional(field("variadic", $.variadic)),
        field("name", $.identifier),
        $._type_annotation
      ),

    parameter_list: ($) => seq("(", commaSep($.parameter_definition), ")"),
```

(also totally okay keeping the current implementation if there are issues with that one)